.editorconfig

```diff
@@ -0,0 +1,2 @@
+[*.lua]
+indent_style = tab
```

Comment:
While we are at it, maybe also:
```
charset = utf-8
trim_trailing_whitespace = true
```

Reply:
I kept it quite conservative as I'm not sure what any existing code styles are, but I'm fully in support of adding some more rules to it like trimming whitespace.

Maybe we might as well add it in this PR also.

I'm hesitant to set it as root though, since people (like me) can have editorconfigs higher up in the hierarchy (in my home folder) that set a bunch of global rules for files that might not be checked in and just used as local development files (say, a Ruby script for example).

Reply:
`charset` and `trim_trailing_whitespace` are now added in the "global" section. I did leave out `root` for the above stated reasons though.